modules/battenberg/1.1/battenberg.smk

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Ryan Morin
+# Module Author:    Ryan Morin
+# Contributors:     N/A
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+import glob
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["battenberg"]`
+CFG = op.setup_module(
+    name = "battenberg",
+    version = "1.1",
+    subdirectories = ["inputs", "infer_sex","battenberg", "outputs"],
+)
+
+#set variable for prepending to PATH based on config
+SCRIPT_PATH = CFG['inputs']['src_dir']
+#this is used in place of the shell.prefix() because that was not working consistently. This is not ideal. 
+
+#this preserves the variable when using lambda functions
+_battenberg_CFG = CFG
+
+# Define rules to be run locally when using a compute cluster
+# TODO: Replace with actual rules once you change the rule names
+localrules:
+    _install_battenberg,
+    _battenberg_input_bam,
+    _battenberg_output_seg,
+    _battenberg_to_igv_seg,
+    _battenberg_cleanup,
+    _battenberg_all,
+
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _battenberg_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.crai"
+    group: "setup_run"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bam + ".bai", output.bai)
+        op.relative_symlink(input.bam + ".bai", output.crai)
+
+# Installs the Battenberg R dependencies and associated software (impute2, alleleCounter)
+# Currently I think this rule has to be run twice for it to work properly because the conda environment is created here. 
+# I am open to suggestions for how to get around this.
+rule _install_battenberg:
+    output:
+        complete = "config/envs/battenberg_dependencies_installed.success"
+    conda:
+        CFG["conda_envs"]["battenberg"]
+    shell:
+        """
+        R -q -e 'devtools::install_github("Crick-CancerGenomics/ascat/ASCAT")' && ##move some of this to config?
+        R -q -e 'devtools::install_github("morinlab/battenberg")' &&              ##move some of this to config?
+        touch {output.complete}"""
+
+# this process is very fast on bam files and painfully slow on cram files. 
```

Comment:
This is a new rule

Reply:
So it doesn't need reference for cram files here?